lib/active_interactor/interactor/worker.rb

```diff
@@ -89,7 +89,8 @@ def execute_interactor_rollback!
       def handle_error(exception)
         @context = interactor.finalize_context!
         execute_rollback
-        raise exception
+        interactor.run_callbacks :failure
+        ActiveSupport::Notifications.instrument('interactor failure') { raise exception }
```

Comment:
I think failure is expected to run before rollback at least that is how is sound like. WDYT?

Reply:
Hmmm that's a valid point... I think in this case there are two concepts of failure here.  There's the broader "An interactor has failed so we need to run rollbacks and then raise an error", and then there is the actual failure (raising the error). We already have callback methods around the rollback process, so really what this callback is is "before_error" however I find that terminology confusing and prefer `before_failure`

Reply:
The goal is not necessarily to capture an interactor's state before rollback, but rather to provide an avenue to allow for things like sending off honeybadger events.

Reply:
actually no you're right I DO want to do this before rollback on second thought, that way we aren't relying on JUST the context.errors, but being able to inspect the objects created.

Reply:
fixed

Reply:
Thank you 🙏